evm_loader/program/src/executor.rs

```diff
@@ -402,12 +421,12 @@ impl<'config, B: Backend> Machine<'config, B> {
 
     #[allow(clippy::too_many_lines)]
     pub fn step(&mut self) -> Result<(), ExitReason> {
-
+        let gas_limit = self.executor.state.block_gas_limit().as_u64();
         match self.step_opcode(){
             RuntimeApply::Continue => { Ok(()) },
             RuntimeApply::Call(info) => {
                 let code = self.executor.code(info.code_address);
-                self.executor.state.enter(u64::max_value(), false);
+                self.executor.state.enter(gas_limit, false);
```

Comment:
Why you are using `block_gas_limit()` there?

Reply:
Wasn't sure if the block gas limit makes sense here. Reverted to u64::MAX.